contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -390,11 +390,16 @@ func (s *SchedulerServer) prepareExecutorInfo(hks hyperkube.Interface) (*mesos.E
 		return nil, fmt.Errorf("either run this scheduler via km or else --executor-path is required")
 	} else {
 		if strings.Index(s.kmPath, "://") > 0 {
-			// URI could point directly to executable, e.g. hdfs:///km
-			// or else indirectly, e.g. http://acmestorage/tarball.tgz
-			// so we assume that for this case the command will always "km"
-			ci.Uris = append(ci.Uris, &mesos.CommandInfo_URI{Value: proto.String(s.kmPath), Executable: proto.Bool(true)})
-			ci.Value = proto.String("./km") // TODO(jdef) extract constant
+			if strings.HasPrefix(s.kmPath, "file://") {
+				// If `kmPath` started with "file://", `km` in agent local path was used.
+				ci.Value = proto.String(s.kmPath)
```

Comment:
"file://" is handled by Mesos Fetcher at https://github.com/apache/mesos/blob/master/src/slave/containerizer/fetcher.cpp#L199